balena/models/device_os.py

```diff
@@ -274,14 +274,24 @@ def get_supported_versions(self, device_type):
                 * default - recommended (if available) or latest otherwise.
 
         """
+        
+        prod_suffix = '.prod'
 
         response = self.base_request.request(
             '/device-types/v1/{device_type}/images'.format(device_type=device_type), 'GET',
             endpoint=self.settings.get('api_endpoint'), auth=False
         )
 
         potential_recommended_versions = [i for i in response['versions'] if not re.search(r'(\.|\+|-)dev', i)]
-        potential_recommended_versions = [i for i in potential_recommended_versions if not semver.parse(i)['prerelease']]
+        
+        # returned versions contain '.prod' suffix (like 2.88.4.prod or 2.87.16+rev1.prod) which is not a valid semver string
+        # we need to remove the suffix before passing to semver.parse()
+        potential_recommended_versions = [
+            i for i in potential_recommended_versions if not semver.parse(
+                i[:-len(prod_suffix)] if i.endswith(prod_suffix) else i
```

Comment:
In balena-semver we would be checking whether there is any `+` before` the `.prod/.dev` suffix, and iff not then change the `.` to a `+` so that's parse-able. But on the other hand we only do that for sorting purposes and the version strings in the `potential_recommended_versions` should be the original ones, not the normalized ones, otherwise the img-maker will not be able to find a matching image on S3.